### PR TITLE
Feature/50 add remote debugging info

### PR DESCRIPTION
**Pull request by JoergZeidler**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -26,5 +27,51 @@
...
+            "type": "cppdbg",
+            "request": "launch",
+            "program": "${workspaceRoot}/gen/cplusplus-service-armv7hf-dev/cplusplus_service",
+            "stopAtEntry": true,
+            "cwd": "${workspaceRoot}",
+            "environment": [],
+            "externalConsole": true,
+            "MIMode": "gdb",
+            "miDebuggerPath": "/usr/bin/gdb-multiarch",
+            "miDebuggerServerAddress": "172.18.18.100:1234",
```
[the remark this answers is not in the document]
Comment (mlilien) [reply .vscode/launch.json:42]: https://stackoverflow.com/questions/29971572/how-do-i-add-environment-variables-to-launch-json-in-vscode

File: .vscode/launch.json [hunk 2]
```diff
@@ -26,5 +27,51 @@
...
+            "type": "cppdbg",
+            "request": "launch",
+            "program": "${workspaceRoot}/gen/cplusplus-service-aarch64-dev/cplusplus_service",
+            "stopAtEntry": true,
+            "cwd": "${workspaceRoot}",
+            "environment": [],
+            "externalConsole": true,
+            "MIMode": "gdb",
+            "miDebuggerPath": "/usr/bin/gdb-multiarch",
+            "miDebuggerServerAddress": "172.18.18.99:1234",
```
[the remark this answers is not in the document]
Comment (mlilien) [reply .vscode/launch.json:66]: https://stackoverflow.com/questions/29971572/how-do-i-add-environment-variables-to-launch-json-in-vscode

File: README.md [hunk 3]
```diff
@@ -192,6 +205,45 @@ Therefore a mapping from  /source to  service must be setup in the sourceFileMap
...
 ```
 
+#### Remote target debugging
+
+This building block provides also a cross-compiled "service hello world example" for aarch64 and armv7 architecture. 
+
+The next steps are exemplary for the aarch64 architecture and show the way to remote debugging on the target.
+
+ **at host system**
+- generate with dobi a service build docker image
```
[the remark this answers is not in the document]
Comment (mlilien) [reply README.md:203]: still ... why dobi? the only precondition is, that it is possible to deploy it to the target. can be build by dobi or ci/cd. I know we don't publish that images via ci/cd... but as a user, I would be confused if I already have a working ci/cd chain.

File: README.md [hunk 4]
```diff
@@ -192,6 +205,45 @@ Therefore a mapping from  /source to  service must be setup in the sourceFileMap
...
+
+```sh
+./dobi.sh cplusplus-service-build-aarch64-dev
+```
+**at target system**
+- store the docker image on your target
```
[the remark this answers is not in the document]
Comment (mlilien) [reply README.md:209]: not done

File: README.md [hunk 5]
```diff
@@ -192,6 +205,45 @@ Therefore a mapping from  /source to  service must be setup in the sourceFileMap
...
+
+- start the docker image on your target
+
+```sh
+docker run --rm -it --privileged --entrypoint /bin/bash -p 1234:1234 xxXXxxXX/bb-cplusplus-service-aarch64-dev
```
[the remark this answers is not in the document]
Comment (mlilien) [reply README.md:214]: it isn't?!

File: README.md [hunk 6]
```diff
@@ -192,6 +205,45 @@ Therefore a mapping from  /source to  service must be setup in the sourceFileMap
...
+```
+
+- start gdbserver in running target container
+
+```sh
+gdbserver :1234 /usr/local/bin/cplusplus_service
+```
+**at host system**
+
+This buildblock provide a visual studio code example debug file (.vscode/launch.json).
```
[the remark this answers is not in the document]
Comment (mlilien) [reply README.md:224]: still open

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,6 +5,9 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+-  added remote target debugging
```
Comment (mlilien) [inline CHANGELOG.md:9]: added a "howto remote target debugging"

File: README.md [hunk 8]
```diff
@@ -116,34 +116,43 @@ For further information how to configure doxygen, see:
 
 ### Using dobi to start debugging
 
-This building block provide a setup to debug a C++ component via remote debugging with a gdbserver.
+This building block provides a setup to debug a x86 C++ component via remote debugging with a gdbserver.
 
-For this purpose a dobi job is provided to do the startup of a gdb server. For demonstration purposes the Hello World service was used as test application. To start the debug session, the following command can be used.
+For this purpose a dobi job is provided to do the startup of a gdbserver. For demonstration purposes, the Hello World service was used as test application. To start the debug session, the following command can be used.
```
Comment (mlilien) [inline README.md:121]: the "Hello World" service

**Review by mlilien — CHANGES_REQUESTED** [review]

File: README.md (hunk 8 above)
Comment (JoergZeidler) [reply README.md]: done

File: CHANGELOG.md (hunk 7 above)
Comment (JoergZeidler) [reply CHANGELOG.md]: done

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (JoergZeidler) [reply README.md]: done

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (JoergZeidler) [reply README.md]: changed

File: README.md [hunk 9]
```diff
@@ -192,6 +205,45 @@ Therefore a mapping from  /source to  service must be setup in the sourceFileMap
...
+ **at host system**
+- generate with dobi a service build docker image
+
+```sh
+./dobi.sh cplusplus-service-build-aarch64-dev
+```
+**at target system**
+- store the docker image on your target
+
+- start the docker image on your target
```
[the remark this answers is not in the document]
Comment (JoergZeidler) [reply README.md]: done

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (JoergZeidler) [reply README.md]: done

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (JoergZeidler) [reply README.md]: doku changed

File: .vscode/launch.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoergZeidler) [reply .vscode/launch.json:42]: No meaningful solution found

File: .vscode/launch.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (JoergZeidler) [reply .vscode/launch.json:66]: No meaningful solution found

File: .vscode/launch.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (mlilien) [reply .vscode/launch.json:42]: for future me:
how to reference env variables: https://code.visualstudio.com/docs/editor/variables-reference#_environment-variables
why there is no "meaningful solution": https://github.com/microsoft/vscode/issues/95371#issuecomment-614725802

**Review by mlilien — CHANGES_REQUESTED** [review]

File: README.md [hunk 10]
```diff
@@ -116,34 +116,43 @@ For further information how to configure doxygen, see:
...
 An example launch.json (.vscode/launch.json) for application cplusplus_service could look like this.
 
 - name: hello-world-x86_64 (remote gdb)
 
-- program: ${workspaceRoot}/gen/cplusplus-service-x86_64-dev/cplusplus_service
+- program: /install/cplusplus_service<br>
+location of the binary file in the docker container
+
+- miDebuggerPath: "${workspaceFolder}/gdb-multiarch.sh"<br>
+location of the gdb-multiarch debugger of your host system
```
[the remark this answers is not in the document]
Comment (JoergZeidler) [reply README.md]: cahnged

File: README.md [hunk 11]
```diff
@@ -116,34 +116,43 @@ For further information how to configure doxygen, see:
...
 
-- miDebuggerServerAddress: localhost:1234
-The Docker container starts a gdbserver at port 1234 and maps it to localhosts port 1234
+- miDebuggerServerAddress: 172.20.0.1:1234<br>
```
[the remark this answers is not in the document]
Comment (JoergZeidler) [reply README.md]: handling changed.

**Review by mlilien — APPROVED** [review]

**Pull request merged**